src/ucp/wireup/wireup_cm.c

```diff
@@ -1068,8 +1095,33 @@ void ucp_cm_server_conn_request_cb(uct_listener_h listener, void *arg,
                      UCT_DEVICE_NAME_MAX);
     memcpy(ucp_conn_request->remote_dev_addr, remote_data->dev_addr,
            remote_data->dev_addr_length);
+
+    /* [ sa_data | ucp_address | user_data ]
+     * |      conn_priv_data_length        |
+     * 1. sa_data is fixed of size, so copy it directly */
     memcpy(&ucp_conn_request->sa_data, remote_data->conn_priv_data,
-           remote_data->conn_priv_data_length);
+           sizeof(ucp_conn_request->sa_data));
+
+    /* 2. Unpack ucp_address and get it's length in buffer */
+    addr_flags                    = ucp_cm_address_pack_flags(worker);
+    ucp_conn_request->remote_addr = ucs_malloc(sizeof(ucp_unpacked_address_t),
+                                               "ucp_unpacked_address_t");
+
```

Comment:
why need to allocate address separately now?

Reply:
Because it'll be a part of the connection request. To save time we unpack remote address at wireup_cm level not at ucp_ep_sever_accept:
https://github.com/openucx/ucx/pull/6859/files/700ca473f677b36c43a26dd5286226700a165d66#diff-820b1decca0c579913725cc65bff6ae38b6b501c5ccb49b2f993f439a91baa9bL757

Reply:
why can't you save an instance of `ucp_unpacked_address_t` not a pointer in `ucp_conn_request_t`?

Reply:
Because `ucp_unpacked_address_t` has a pointer to a list of addresses and compiler doesn't know it's layout, so fails: `error: field ‘remote_addr’ has incomplete type`